### ISSUE TITLE
Cache manager is not working at all

### ISSUE THREAD

**Issue by svensklave** [lexich/redux-api#135]
Is there any plans to fix cache manager?

**Comment by lexich** [comment]
Can you explain you case with code?

**Comment by svensklave** [comment]
Hi!
I tried to do in various ways to get cache to work as expected
like: 
one endpoint with different params (http://***.com/api/v1/:type/:resolution)
so I added in configuration a cache object with expire parameter set to 360

```
const apitest = dataLayer({
  posts: {
    url: 'https://jsonplaceholder.typicode.com/:posts/:id',
    cache: {
      expire: 10
    },
    options: {
      headers: {
        Accept: 'application/json'
      }
    }
  }
}).use('fetch', adapterFetch);
```
so this adds cache and everything is ok except that it anyway does requests to 
url endpoint (something wrong with checking the cache time with date)

seems this part is little wrong (src/utils/cache.js)
```
export function setExpire(value, oldDate) {
  let expire = value;
  if (oldDate instanceof Date) {
    if (typeof expire === "number" || expire instanceof Number) {
      const d = MockNowDate.pop();
      d.setSeconds(d.getSeconds() + expire);
      expire = d;
    }
    if (expire instanceof Date) {
      if (expire.valueOf() < oldDate.valueOf()) {
        expire = oldDate;
      }
    }
  }
  return expire;
}
```
and this looks also strange (src/reducerFn.js):
```
...
      case actionCache:
        const { id, data } = action;
        const expire = setExpire(action.expire, state.cache.expire);
        return {
          ...state,
          cache: { ...state.cache, [id]: { expire, data } }
        };
...
```
so setExpire(action.expire, state.cache.expire);
state.cache.expire
undefined because
cache looks like this
```
data(pin): "posts_posts=posts;id=2;"
cache:{
posts_posts=posts;id=2;: { expire: 10, data: {…} }
posts_id=1;posts=posts;: { expire: 10, data: {…} }
}
error(pin): null
```
shouldnt this be like this:
setExpire(action.expire, state.cache[id].expire);

but anyway this not helps because of
setExpire function
and also getData function looks weird

when I did these changes then everything works good:
of course these changes fits my needs so maybe this code could be extended to be more
flexible but ok here goes:
utils/cache.js
```
export function setExpire(value) {
  let expire = value;
  if(expire){
    const d = new Date();
    d.setSeconds(d.getSeconds() + expire);
    expire = d;
  }
  return expire;
};
```
actionFn.js
```
...
fetch = (...){
...
//changed this
/* 
      const data = cacheManager.getData(
        cache && id && cache[id] !== undefined && cache[id]
      );
      if (data !== undefined) {
        return Promise.resolve(data);
      }
*/
// to this
      const isDataCached = cacheManager.isExpired(
        cache && id && cache.hasOwnProperty(id) && cache[id].expire
      );
      if (isDataCached) {
        return Promise.resolve(cache[id].data);
      }
...
}
```
utils/cache.js
where Manager = {...}
```
// changed this
/*
  getData(cache) {
    if (!cache) { return; }
    const { expire, data } = cache;
    if (expire === false || expire === undefined || expire === null) {
      return data;
    }
    if (expire instanceof Date) {
      if (expire.valueOf() > MockNowDate.pop().valueOf()) {
        return data;
      }
    }
  },
*/
// to this
  isExpired(expire) {
    if (!expire) { return; }
    if (expire instanceof Date) {
      return expire.valueOf() > (new Date()).valueOf();
    }
  },
```
reducerFn.js
cache reducer 
```
      const { id, data } = action;
      const expire = setExpire(action.expire);
      return {
        ...state,
        cache: { ...state.cache, [id]: { expire, data } }
      };
```

sorry about my english and style (and not much time to sit at pc on my vacation :) )

I hope I did not forget some stuff
anyway maybe it helps

**Comment by svensklave** [comment]
and one more thing 
coudl anybody explain why there is need for this 
just want to know the idea/logics - maybe I am missing something so wan to learn if there is some 
reason behind this:
utils/cache.js
```
export const MockNowDate = {
  date: undefined,
  push(date) {
    this.date = date;
  },
  pop() {
    if (this.date) {
      const d = this.date;
      this.date = undefined;
      return new Date(d);
    } else {
      return new Date();
    }
  }
};
```

**Comment by lexich** [comment]
@svensklave About `MockNowDate` - its wrapper around `new Date`. This construction allow to write unit tests.
```js
var date = new Date('fixed date');
MockNowDate.push(date);
// now in code MockNowDate.pop() return date.
```

**Comment by lexich** [comment]
About case, it's very strange, but I try to write correct test to catch this situation

**Comment by lexich** [comment]
@svensklave Sorry for long delay. I fixed bug with expire, pls check version 0.10.4
